### PR TITLE
Promote PageHeader to Beta 🎉

### PR DESCRIPTION
**Pull request by broccolinisoup**
We completed the PageHeader work and closed out [the epic (internal link only)](https://github.com/github/primer/issues/795) 

#### Changed

- Move PageHeader to Beta status and update the component accordingly.

### Rollout strategy

- [ ] Patch release
- [x] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### CODE REVIEW

File: packages/react/src/PageHeader/PageHeader.docs.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
   "id": "drafts_page_header",
   "name": "PageHeader",
-  "status": "draft",
+  "status": "beta",
```
Comment (broccolinisoup) [inline packages/react/src/PageHeader/PageHeader.docs.json:4]:
```suggestion
  "status": "alpha",
```

File: packages/react/src/PageHeader/PageHeader.docs.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
 {
   "id": "drafts_page_header",
   "name": "PageHeader",
-  "status": "draft",
+  "status": "alpha",
```
Comment (broccolinisoup) [inline packages/react/src/PageHeader/PageHeader.docs.json:4]: Although we have usages in production (though I think they are only staff shipped), I think it is better to promote to Alpha first until we develop our confidence in the API. Let me know if you have any concerns 🙌🏻

File: .changeset/nice-forks-joke.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@primer/react": minor
+---
+
+PageHeader: Promote the component to Beta status and now it is available to import from "@primer/react"
```
Comment (broccolinisoup) [inline .changeset/nice-forks-joke.md:5]:
```suggestion
PageHeader: Promote the component to Alpha status and now it is available to import from "@primer/react"
```

**Review by joshblack — APPROVED** [review]
🎉

**Review by siddharthkp — APPROVED** [review]
LGTM! 

Could even promote to beta

**Pull request merged**